packages/shylu/shylu_node/basker/src/shylubasker_nfactor_blk.hpp

```diff
@@ -1387,27 +1416,29 @@ namespace BaskerNS
         const Int pend = L.col_ptr(t+1-local_offset);
         const Entry xj = X(j);
 
-        //We could use kokkos local memory,
-        //to get rid of this pragma
-        //
-        //What we want to think about is how to 
-        //do this in a dense way
-        //#pragma ivdep (may be slower)
-        //printf( " gperm(%d+%d) = %d, scol=%d, %d:%d\n",(int)j,(int)brow,(int)t, (int)L.scol, (int)L.col_ptr(t-local_offset)+1,(int)pend );
-        for(Int p = L.col_ptr(t-local_offset)+1; p < pend; ++p)
-        {
-          const Int row_idx = L.row_idx(p);
-          const Entry update_val = L.val(p)*xj;
-
-          //printf( " %d: row_idx=%d\n",(int)p,(int)row_idx );
-          X(row_idx) -= update_val;
+        if (xj != zero) {
```

Comment:
Looks like `xj` was unused prior to adding this check. If `xj` is zero why do we bypass this block of code?